acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -228,7 +258,209 @@ impl<'a, F: AcirField, B: BlackBoxFunctionSolver<F>> VM<'a, F, B> {
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                            rhs_value = match rhs.expect_u32() {
+                                Ok(rhs_value) => rhs_value.into(),
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                        }
+                        IntegerBitSize::U64 => {
+                            lhs_value = match lhs.expect_u64() {
+                                Ok(lhs_value) => lhs_value.into(),
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                            rhs_value = match rhs.expect_u64() {
+                                Ok(rhs_value) => rhs_value.into(),
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                        }
+                        IntegerBitSize::U128 => {
+                            lhs_value = match lhs.expect_u128() {
+                                Ok(lhs_value) => lhs_value,
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                            rhs_value = match rhs.expect_u128() {
+                                Ok(rhs_value) => rhs_value,
+                                Err(..) => {
+                                    return;
+                                }
+                            };
+                        }
+                    };
+                    let c = match result {
+                        MemoryValue::U1(value) => value,
+                        _ => {
+                            return;
+                        }
+                    };
+                    let approach_index = self.branch_to_feature_map[&(
+                        self.program_counter,
+                        usize::MAX
+                            - 2
+                            - rhs_value
+                                .abs_diff(lhs_value)
+                                .checked_ilog2()
+                                .map_or_else(|| 0, |x| x + 1)
+                                as usize,
+                    )];
+                    let condition_index = self.branch_to_feature_map
+                        [&(self.program_counter, if c { usize::MAX } else { usize::MAX - 1 })];
+                    self.fuzzer_trace[condition_index] += 1;
+                    self.fuzzer_trace[approach_index] += 1;
+                }
+            }
+        }
+    }
+    pub fn get_fuzzing_trace(&self) -> Vec<u32> {
+        if !self.fuzzing_active {
+            Vec::new()
```

Comment:
I think in this case `self.fuzzer_trace` will be an empty Vec so we could simplify this method to always return `self.fuzzer_trace.clone()`

Reply:
Right